pkg/daemon/ceph/osd/volume_test.go

```diff
@@ -260,6 +268,236 @@ var cephVolumeRAWTestResult = `{
 }
 `
 
+func TestInitializeBlockPVC(t *testing.T) {
+	executor := &exectest.MockExecutor{}
+	executor.MockExecuteCommandWithCombinedOutput = func(command string, args ...string) (string, error) {
+		logger.Infof("%s %v", command, args)
+		if args[1] == "ceph-volume" && args[2] == "raw" && args[3] == "prepare" && args[4] == "--bluestore" {
+			return initializeBlockPVCTestResult, nil
+		}
+
+		return "", errors.Errorf("unknown command %s %s", command, args)
+	}
+
+	context := &clusterd.Context{Executor: executor}
+	clusterInfo := &cephconfig.ClusterInfo{
+		CephVersion: cephver.CephVersion{Major: 14, Minor: 2, Extra: 8},
+	}
+	a := &OsdAgent{cluster: clusterInfo, nodeName: "node1"}
+	devices := &DeviceOsdMapping{
+		Entries: map[string]*DeviceOsdIDEntry{
+			"data": {Data: -1, Metadata: nil, Config: DesiredDevice{Name: "/mnt/set1-data-0-rpf2k"}},
+		},
+	}
+
+	blockPath, metadataBlockPath, err := a.initializeBlockPVC(context, devices, false)
+	assert.Nil(t, err)
```

Comment:
You also need another test that asserts NotNil when 14.2.7 is used which means the provision Executor is failing since the CLI is different.